### PR TITLE
added regression example and fixed script issue

### PR DESCRIPTION
**Pull request by yimingjiang**
@christos-c @danyaljj 

I have fixed the script issue and added the regression example. Code are documented and data set is cited and documented. All other examples compiled successfully.

### CODE REVIEW

**Comment by yimingjiang** [comment]
@christos-c @danyaljj 

I wonder why semaphore build failed. My guess is that to compile all examples, the script is now `sh compileLBJ.sh all` rather `sh compileLBJ.sh`.

**Comment by danyaljj** [comment]
I have invited you to the Semaphore. You can login and see the log.

**Comment by yimingjiang** [comment]
@christos-c @danyaljj It turns out that I need to add the license header for code. Thanks for changing the script on Semaphore @danyaljj . Now build succeeded.

**Comment by danyaljj** [comment]
Thanks @yimingjiang I will go through the details.

File: data/regression/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Regression Data Set
```
Comment (danyaljj) [inline data/regression/README.md]: Could you add a pointer to your Regression example in the tutorial?

**Comment by danyaljj** [comment]
Looks good to me. Passing the torch to @christos-c .

File: lbjava-examples/target/classes/.gitignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Ignore everything in this directory
```
Comment (christos-c) [inline lbjava-examples/target/classes/.gitignore]: We shouldn't track any files inside `target`.

**Comment by christos-c** [comment]
@yimingjiang looks good. Once you address mine and @danyaljj issues he can merge.

File: lbjava-examples/target/classes/.gitignore (hunk 2 above)
Comment (yimingjiang) [inline lbjava-examples/target/classes/.gitignore]: I was trying to add an empty directory `classes` under `target`. Git can't just add an empty directory.

Comment (christos-c) [inline lbjava-examples/target/classes/.gitignore]: If you see #40 I have addressed this problem.

File: data/regression/README.md (hunk 1 above)
Comment (yimingjiang) [inline data/regression/README.md]: I have added a line with pointers to the tutorial.

**Comment by yimingjiang** [comment]
@danyaljj @christos-c I have resolved issues and updated with the repo. Also I have checked that the example runs as expected. Please review the changes.

**Pull request merged**